Compiling the kubelet in providerless mode is broken for release-1.14 & release-1.15

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**: Compiling the kubelet in providerless mode fails with

```shell
# k8s.io/kubernetes/pkg/cloudprovider/providers/azure
pkg/cloudprovider/providers/azure/azure_controller_common.go:66:13: undefined: lockMap
```

This is the issue for both
- [release-1.14](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-current-minus-2/builds/259)
- [release-1.15](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-current-minus-1/builds/254)

**What you expected to happen**: No compilation error

**How to reproduce it (as minimally and precisely as possible)**:

```shell
mkdir /tmp/lockless && cd /tmp/lockless
git clone https://github.com/kubernetes/kubernetes && cd kubernetes
GOFLAGS=-tags=providerless KUBE_VERBOSE=0 KUBE_BUILD_PLATFORMS=linux/amd64 ./build/run.sh make all WHAT=cmd/kubelet
# ==> fails
```

**Anything else we need to know?**:

Those are most probably the CPs that have introduced the issue:
- release-1.14: https://github.com/kubernetes/kubernetes/pull/85832
- release-1.15: https://github.com/kubernetes/kubernetes/pull/85830

[release-1.16](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-current/builds/255) & [master](https://hush-house.pivotal.io/teams/k8s-c10s/pipelines/kind/jobs/kind%20master%20with%20k8s-master/builds/262) seem to be fine.

The original PR which has been cherry-picked into the releases is https://github.com/kubernetes/kubernetes/pull/85115.

/assign @aramase @andyzhangx
/kind bug
/sig cloud-provider
/area provider/azure

**Environment**:
- Kubernetes version (use `kubectl version`): release-1.1{4,5}
- Cloud provider or hardware configuration: providerless
- OS (e.g: `cat /etc/os-release`): (compiling on) Darwin (via docker) and Linux
- Kernel (e.g. `uname -a`): -
- Install tools: -
- Network plugin and version (if this is a network-related bug): -
- Others: -

--- a/pkg/cloudprovider/providers/azure/azure_utils.go
+++ b/pkg/cloudprovider/providers/azure/azure_utils.go
@@ -1,5 +1,3 @@
-// +build !providerless
-
 /*
 Copyright 2019 The Kubernetes Authors.
 

--- a/pkg/cloudprovider/providers/azure/azure_utils_test.go
+++ b/pkg/cloudprovider/providers/azure/azure_utils_test.go
@@ -1,5 +1,3 @@
-// +build !providerless
-
 /*
 Copyright 2019 The Kubernetes Authors.
 

